### DISCUSSION TITLE
$wire magic inside if statement is null/undefined

### DISCUSSION THREAD

**Discussion by lukas-frey** [alpinejs/alpine#2446] [7. Bugs (archived)]
If you wrap an if statement / ternary operator around a `$wire` magic property inside an `x-on:xxx` directive, Alpine throws the following errors:
```
Alpine: Cannot reference "$wire" outside a Livewire component.
Alpine Expression Error: null is not an object (evaluating 'wireEl.__livewire')
```
The first error is weird. I used it inside a livewire component, removing the if statement worked. But obviously, not as intended (it needs to run only if the condition is true).

A short snippet of the code I tried to run: (not working)
```
...
<x-foo
x-data="{show: true}"
x-on:animationend="if (!show) $wire.baz"
...
x-on:click="show = false"
>
Bar
</x-foo>
...
```

This however works:
```
...
<x-foo
x-data="{show: true}"
x-on:animationend="$wire.baz"
...
x-on:click="show = false"
>
Bar
</x-foo>
...
```

**Comment by SimoTod** [comment]
$wire is not part of Alpine, it's implemented in the Livewire repo.
The first step would be to look in that repo and see when `Cannot reference "$wire" outside a Livewire component.` is triggered.

**Comment by SimoTod** [comment]
and please, wait for people to see the discussion before opening an issue that might not even be related to this repo. :)
